### ISSUE TITLE
techdocs page refresh twice when switching to a new page

### ISSUE THREAD

**Issue by falconbin** [backstage/backstage#5487]
we use backstage\techdocs to host a document, but when switch to a new page, the page refresh twice: one fresh for the old page, then again fresh for the new page. Same issue on backstage demo: https://demo.backstage.io/docs/default/Component/backstage/overview/what-is-backstage/

## Expected Behavior
should only refresh only once.

## Current Behavior
the page refresh twice instead of once

## Possible Solution
reason: when switches to a new page, the old page refresh because the shadowDom reloaded.
 fix: add check to avoid dom reload for the old page
Code change as below(./node_modules/@backstage/plugin-techdocs/dist/index.esm.js):
![image](https://user-images.githubusercontent.com/20329141/116190272-e83be000-a75c-11eb-98d7-8635a5c271b5.png)

## Steps to Reproduce
Step1: login to: https://demo.backstage.io/docs/default/Component/backstage/overview/what-is-backstage/
Step2: switch to page 'Backstage architecture' by clicking on it.
Step3: you'll find that it refreshes twice

## Your Environment

- NodeJS Version (v12):
- Operating System and Version (e.g. Ubuntu 14.04):
- Browser Information:

**Comment by Fox32** [comment]
I wonder if this is a duplicate of #5184 which was fixed recently. Can you still reproduce it in master? The fix wasn't released yet and is not available on the demo page yet.

**Comment by falconbin** [comment]
> I wonder if this is a duplicate of #5184 which was fixed recently. Can you still reproduce it in master? The fix wasn't released yet and is not available on the demo page yet.

The issue should be fixed in master branch as I can't reproduce it, could be closed. Thanks, @Fox32
